lib/styles.js

```diff
@@ -180,4 +197,90 @@ module.exports = function (expect) {
+                            if (j > 0) {
+                                this.nl();
+                            }
+                            if (lineEntry.style === 'text') {
+                                var styles = lineEntry.args.styles;
+                                if (pen && styles.length === 1 && typeof pen[styles[0]] === 'function') {
+                                    // Text with a single style also available as a method on the pen being inspected:
+                                    this
+                                        .text('.')
+                                        .jsFunctionName(styles[0])
+                                        .text('(')
+                                        .singleQuotedString(lineEntry.args.content)
+                                        .text(')');
+                                } else {
+                                    this
+                                        .text('.')
+                                        .jsFunctionName('text')
+                                        .text('(')
+                                        .singleQuotedString(lineEntry.args.content);
+                                    if (styles.length > 0) {
+                                        this
+                                            .text(', ')
+                                            .appendInspected(styles.length === 1 && Array.isArray(styles[0]) ? styles[0] : styles);
+                                    }
+                                    this.text(')');
+                                }
+                            } else {
+                                // lineEntry.style === 'block'
+                                this
+                                    .text('.')
+                                    .jsFunctionName('block').text('(').jsKeyword('function').text(' () {');
+                                if (lineEntry.args) {
+                                    this
+                                        .nl()
+                                        .indentLines()
+                                        .i()
+                                        .magicPen(pen, lineEntry.args)
+                                        .outdentLines()
+                                        .nl();
+                                }
+                                this.text('})');
+                            }
+                        }, this);
+                    }, this);
+                    if (!isTopLevel) {
+                        this.text(';');
+                    }
+                });
+
+                this.outdentLines();
+            }
+        });
+
+        // If we're at the top level of a non-empty pen compatible with the current output,
+        // render the output of the pen in a comment:
+        if (isTopLevel && !pen.isEmpty() && (pen.format === this.format || !pen.format)) {
+            this.sp().commentBlock(pen);
+        }
+    });
 };
```

Comment:
I would prefer to split this into functions that would handle a line and a function that handle multiple lines. That would remove some of the nesting.